index.js

```diff
@@ -143,8 +143,40 @@ class Dynamodb extends Datastore {
         });
     }
 
+    /**
+     * Scan records in the datastore
+     * @method scan
+     * @param  {Object}   config                Configuration object
+     * @param  {String}   config.table          Table name
+     * @param  {Object}   [config.params]       Record data
+     * @param  {Object}   config.paginate       Pagination parameters
+     * @param  {Number}   config.paginate.count Number of items per page
+     * @param  {Number}   config.paginate.page  Specific page of the set to return
+     * @param  {Function} callback              fn(err, data)
+     *                                          err - Error object
+     *                                          data - Data saved in the table
+     */
     scan(config, callback) {
-        callback();
+        const client = this.client[config.table];
+        const endIndex = config.paginate.page * config.paginate.count;
+        const startIndex = (config.paginate.page - 1) * config.paginate.count;
+
+        if (!client) {
+            const err = new Error(`Invalid table name "${config.table}"`);
+
+            return callback(err);
+        }
+
+        return client.scan().limit(endIndex)  // scan all items up to and including last item specified
```

Comment:
If `client.scan` includes the last item then it should be `(endIndex-1)` here

For example, the records are `[a,b,c,d,e,f]`
`page:2 count:2` should give `c,d`
Start index = `(2-1)*2 = 2`
End index = `2*2 = 4`